### PR TITLE
Button tweaks

### PR DESCRIPTION
**Pull request by annr**

### CODE REVIEW

**Comment by unindented** [comment]
Ah sorry, didn't see the `wip` label! Ignore my approval.

**Comment by annr** [comment]
I love the approval!

I have new visual diffs, and support for icons on the right, but I'm going to try to get sign-off from Adrian before I commit all that. Although maybe I should check in the new code for approval...

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -20,6 +20,12 @@ export const ButtonSize = Enum({
 });
 export type ButtonSize = Enum<typeof ButtonSize>;
 
+export const IconPosition = Enum({
```
Comment (swese44) [inline src/components/Button/index.tsx]: Do you want to use string literal types for simple options like `right|left`? We have them in a few other places:
https://github.com/Microsoft/YamUI/blob/master/src/components/Block/index.tsx#L35
https://github.com/Microsoft/YamUI/blob/master/src/components/HorizontalList/horizontal-list.tsx#L9

Comment (annr) [reply src/components/Button/index.tsx]: Consistency is good, and I think I like this better anyway. I'l make this change after I get CI to pass.

**Pull request merged**